### PR TITLE
bug/EN-2910-Fix-bug-when-nil-hash-is-return-from-sync-map

### PR DESCRIPTION
**Pull request by SebastianMarian**

### CODE REVIEW

File: process/sync/baseSync.go [hunk 1]
```diff
@@ -425,17 +425,27 @@ func (boot *baseBootstrap) ShouldSync() bool {
 }
 
 func (boot *baseBootstrap) removeHeaderFromPools(header data.HeaderHandler) []byte {
-	value, _ := boot.headersNonces.Get(header.GetNonce())
 	boot.headersNonces.Remove(header.GetNonce())
 
-	hash, ok := value.([]byte)
-	if ok {
-		boot.headers.Remove(hash)
+	hash, err := boot.computeHeaderHash(header)
```
Comment (iulianpascalau) [inline process/sync/baseSync.go]: You can make use of core.CalculateHash

File: process/sync/baseSync.go [hunk 2]
```diff
@@ -425,17 +425,27 @@ func (boot *baseBootstrap) ShouldSync() bool {
...
+	if err != nil {
+		log.Info(err.Error())
+		return nil
 	}
 
+	boot.headers.Remove(hash)
 	return hash
 }
 
+func (boot *baseBootstrap) computeHeaderHash(headerHandler data.HeaderHandler) ([]byte, error) {
```
Comment (sasurobert) [inline process/sync/baseSync.go]: use core.ComputeHash instead

File: process/sync/metablock.go [hunk 3]
```diff
@@ -498,9 +498,9 @@ func (boot *MetaBootstrap) requestHeader(nonce uint64) {
 	}
 }
 
-// getHeaderWithNonce method gets the header with given nonce from pool, if it exist there,
+// getRequestedHeaderIfMissing method gets the header with given nonce from pool, if it exist there,
 // and if not it will be requested from network
-func (boot *MetaBootstrap) getHeaderRequestingIfMissing(nonce uint64) (*block.MetaBlock, error) {
+func (boot *MetaBootstrap) getRequestedHeaderIfMissing(nonce uint64) (*block.MetaBlock, error) {
```
Comment (iulianpascalau) [inline process/sync/metablock.go]: May revert this change: it sounds better "getHeaderRequestingIfMissing"

File: process/sync/shardblock_test.go [hunk 4]
```diff
@@ -2135,10 +2136,17 @@ func TestBootstrap_ForkChoiceIsEmptyCallRollBackOkValsShouldWork(t *testing.T) {
...
+	hasher := &mock.HasherStub{
+		ComputeCalled: func(s string) []byte {
+			return currentHdrHash
+		},
+	}
 
 	//a marshalizer stub
 	marshalizer := &mock.MarshalizerStub{
+		MarshalCalled: func(obj interface{}) ([]byte, error) {
+			return json.Marshal(obj)
```
Comment (iulianpascalau) [inline process/sync/shardblock_test.go]: You are using the gin implementation. Works but you'd better make use of the golang's impl.

File: process/sync/baseSync.go (hunk 2 above)
Comment (SebastianMarian) [reply process/sync/baseSync.go]: Done

File: process/sync/baseSync.go (hunk 1 above)
Comment (SebastianMarian) [reply process/sync/baseSync.go]: Done

File: process/sync/metablock.go (hunk 3 above)
Comment (SebastianMarian) [reply process/sync/metablock.go]: Done

File: process/sync/shardblock_test.go (hunk 4 above)
Comment (SebastianMarian) [reply process/sync/shardblock_test.go]: Removed
